gc/base/SparseVirtualMemory.hpp

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup GC_Base_Core
+ */
+
+#if !defined(SPARSEVIRTUALMEMORY_HPP_)
+#define SPARSEVIRTUALMEMORY_HPP_
+
+#include "omrcomp.h"
+#include "omrport.h"
+#include "modronbase.h"
+#include "modronopt.h"
+
+#include "BaseVirtual.hpp"
+#include "VirtualMemory.hpp"
+
+class MM_GCExtensions;
+class MM_GCExtensionsBase;
+class MM_Heap;
+class MM_SparseAddressOrderedFixedSizeDataPool;
+struct J9PortVmemParams;
+
+/**
+ * @todo Provide class documentation
```

Comment:
Give some description like:

Large virtual memory for allocation of large objects (their data portion). It's sparsely committed only for live objects (memory is eagerly committed/de-committed on allocate/free).
It also maintains mapping from data portion to proxy (heap) object.
The reverse mapping is not maintained (it's expected that heap object has an easy way to find data portion, for example, via pointer in the object header).